Foundation/NSLock.swift

```diff
@@ -36,24 +47,51 @@ open class NSLock: NSObject, NSLocking {
         pthread_mutex_destroy(mutex)
         mutex.deinitialize()
         mutex.deallocate(capacity: 1)
+#if os(OSX) || os(iOS)
+        deallocateTimedLockData(_timedLockData)
+#endif
     }
     
     open func lock() {
         pthread_mutex_lock(mutex)
     }
-    
+
     open func unlock() {
         pthread_mutex_unlock(mutex)
+#if os(OSX) || os(iOS)
+        // Only use _timedLockData if it was initialized by lock(before:)
+        if let tld = _timedLockData {
+            pthread_mutex_lock(_CFGetTimedLockMutex(tld))
+            pthread_cond_broadcast(_CFGetTimedLockCond(tld))
```

Comment:
- why do we use `pthread_cond_broadcast` instead of `pthread_cond_signal`
 - all the return values aren't checked (it's fine to `precondition(ret == 0)`)

Reply:
there may be multiple waiters on a given NSLock so broadcast is the correct idiom

Reply:
@phausler please correct me if I'm wrong but AFAIK _any_ of the waiters will be able to make progress and also after every unlock we run that code. So why do we need to wake up _all_ and not just one?